File: apps/frontend/types/document.ts
/**
 * Represents a document.
 */
export interface Document {
  /** The unique identifier for the document. */
  id: string;
  /** The name of the document file. */
  file_name: string;
  /** The timestamp when the document was uploaded. */
  created_at: string;
  /** The path to the document in Google Cloud Storage. */
  gcs_path: string;
  /** The ID of the user who uploaded the document. */
  user_id: string;
  /** The processing status of the document. */
  status: string;
  /** An optional error message if processing failed. */
  processingError?: string;
}

File: apps/backend/models/document.go
package models

import "time"

type Document struct {
	ID              string    `json:"id"`
	UserID          string    `json:"user_id"`
	FileName        string    `json:"file_name"`
	GCSPath         string    `json:"gcs_path"`
	Status          string    `json:"status"` // e.g., "processing", "processed", "failed"
	ProcessingError string    `json:"processingError,omitempty"`
	CreatedAt       time.Time `json:"created_at"`
}
